docs/coverpoints.md

```diff
@@ -0,0 +1,151 @@
+<!--
+  ~ SPDX-License-Identifier: MIT
+  ~ Copyright (c) 2023-2024 Vypercore. All Rights Reserved
+  -->
+
+## Coverpoints
+
+A coverpoint consists of one or more axes, which are then crossed. (An axis covers one signal/data - similar to a UVM coverpoint). Each possible combination of the axes' values is called a bucket. Each bucket has a default goal with a target of 10 hits, which can be modified as required, or can be made illegal or ignored.
+
+Each new coverpoint should inherit from the Coverpoint class, and requires a 'name' and a 'description'. More arguments can be passed to the coverpoint as required (see example files).
+
+``` Python
+class MyCoverpoint(Coverpoint):
+    def __init__(self, name: str, description: str):
+        super().__init__(name, description)
+```
+
+The above `__init__` method is optional if you are only providing a name and description when instancing. If you wish to create a coverpoint which relies on other data (such as a pre-processed list of instruction names, etc), then further arguments can be provided. These should all be placed before `super().__init_()` if you want them to be accessible to the `setup()` method.
+
+The example below shows instancing the same coverpoint twice with different names. These extra parameters could come from automatically generated lists which split up the original data into smaller groups.
+```Python
+class ChewToysByName(Coverpoint):
+    def __init__(self, name: str, description: str, names):
+        self.name_group = names
+        super().__init__(name, description)
+
+    def setup(self, ctx):
+        self.add_axis(
+            name="name",
+            values=self.name_group,
+            description="Most important dog names only",
+        )
+...
+# To be instanced as below:
+class DogsAndToys(Covergroup):
+    def setup(self, ctx):
+        self.add_coverpoint(
+            ChewToysByName(
+                name="chew_toys_by_name__group_a",
+                description="Preferred chew toys by name (Group A)",
+                names=["Barbara", "Connie", "Graham"],
+            )
+        )
+        self.add_coverpoint(
+            ChewToysByName(
+                name="chew_toys_by_name__group_b",
+                description="Preferred chew toys by name (Group B)",
+                names=["Clive", "Derek", "Ethel"],
+            )
+        )
+```
```

Comment:
I think this might be better being moved to the bottom of the page, after the basics of a coverpoint have been explained. 
This can be under a separate topic which demonstrates how to use multiple instances of a coverpoint